types/ArweaveTransactionID.ts

```diff
@@ -0,0 +1,44 @@
+import { Equatable } from './equatable';
+
+const txIdRegex = /^(\w|-){43}$/;
+export class ArweaveTransactionID implements Equatable<ArweaveTransactionID> {
+  constructor(private readonly transactionId: string) {
+    if (!transactionId.match(txIdRegex)) {
+      throw new Error(
+        'Transaction ID should be a 43-character, alphanumeric string potentially including "-" and "_" characters.',
+      );
+    }
+  }
+
+  [Symbol.toPrimitive](hint?: string): string {
+    if (hint === 'number') {
+      throw new Error('Transaction IDs cannot be interpreted as a number!');
+    }
+
+    return this.toString();
+  }
+
+  toString(): string {
+    return this.transactionId;
+  }
+
+  valueOf(): string {
+    return this.transactionId;
+  }
+
+  equals(entityId: ArweaveTransactionID): boolean {
+    return this.transactionId === entityId.transactionId;
+  }
+
+  toJSON(): string {
+    return this.toString();
+  }
```

Comment:
do we need these?

Reply:
Probably not, can take them out if you want, but I see no reason to not leave them in. I tried looking thru the ardrive core to see where they are used but its proving difficults. I can see myself using `.equals()` to check if a transaction is in the registry or if they are the owner of a contracts. toJSON()....im not sure it useful - theres this line `	[Symbol.toPrimitive](hint?: string): string {
		if (hint === 'number') {
			throw new Error('Transaction IDs cannot be interpreted as a number!');
		}

		return this.toString();
	}` that seems to automatically turn the txid into a string... what do you think?

Reply:
i think leave `equals` and `toString` - we can always add more later but I don't anticipate needing them

Reply:
removed